File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.kt
class Solution {
    fun totalMoney(n: Int): Int {
        val (q: Int, rem: Int) = Pair(n / 7, n % 7)
        
        val total: Int = (28 * q) + 7 * ((q * (q - 1)) / 2) + rem * q + (rem * (rem + 1)) / 2; 

        return total;
    }
}

File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.cpp
class Solution {
public:
    int totalMoney(int n) {
        int q = n / 7, rem = n % 7;
        
        int total = (28 * q) + 7 * ((q * (q - 1)) / 2) + rem * q + (rem * (rem + 1)) / 2; 

        return total;
    }
};